### PR TITLE
feat: add shellcheck and shfmt

### PR DESCRIPTION
**Pull request by vprashar2929**
This PR addresses #88 and #87:
* Adds `shfmt` and `shellcheck` to format and lint shell scripts.
* Modify the existing GH workflow to run `fmt` and `shellcheck` jobs.
* Update the existing scripts accordingly.
* Removes `wait-for-csv.sh` as it is not used anymore.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -58,6 +58,11 @@ pkg/rest/zz_docs: $(wildcard pkg/rest/*.go pkg/korrel8r/*.go) $(SWAG)
 lint: $(VERSION_TXT) $(GOLANGCI_LINT) ## Run the linter to find and fix code style problems.
 	$(GOLANGCI_LINT) run --fix
 
+.PHONY: fmt
```
Comment (alanconway) [inline Makefile:61]: go fmt is not necessary. `golangci-lint run --fix` includes the actions of `go fmt` and is faster than running separately.

File: Makefile [hunk 2]
```diff
@@ -58,6 +58,11 @@ pkg/rest/zz_docs: $(wildcard pkg/rest/*.go pkg/korrel8r/*.go) $(SWAG)
...
+fmt: $(SHFMT) ## Run go fmt against code.
+	go fmt ./...
+	$(SHFMT) -l -w ./**/*.sh
```
Comment (alanconway) [inline Makefile:64]: Suggest running $(SHFMT) as  part of the lint: target. That will simplify the git actions and lets devs ensure they are "lint and format clean" with one command.

File: hack/copyright.sh [hunk 3]
```diff
@@ -4,5 +4,5 @@
 
 LINE="// Copyright: This file is part of korrel8r, released under https://github.com/korrel8r/korrel8r/blob/main/LICENSE"
 
-FILES=$(find -name *.go | xargs grep -L "$LINE")
-if [ -n "$FILES" ]; then sed -i '1 i \'"$LINE"'\n' $FILES; fi
+FILES=$(find . -name '*.go' -print0 | xargs grep -L "$LINE")
```
Comment (alanconway) [inline hack/copyright.sh:7]:
```suggestion
FILES=$(find . -name '*.go' -print0 | xargs -0 grep -L "$LINE")
```

**Review by alanconway — CHANGES_REQUESTED** [review]
Looks good, a couple of suggestions and I'll merge it.

**Comment by alanconway** [comment]
@vprashar2929 There are some very minor clashes with latest main (my changes, sorry!), can you please:
- Squash the PR to a single commit
- Rebase on latest main
This project aims at keeping  a linear history, so rebase is preferred over merge.

**Comment by vprashar2929** [comment]
No problem Alan 😊.  All green now

**Comment by vprashar2929** [comment]
@alanconway Can you help merge this PR? I think auto-merge is struck on the build check

**Comment by alanconway** [comment]
Manually pushed as commit 0a4bf66.
Restored build.xml to original - single "build" job with multiple steps. 
shfmt is run via "make lint" so no need for a separate job.

**Pull request closed without merging**